### PR TITLE
[RF] Add analytical integral for RooLandau

### PR DESCRIPTION
**Pull request by rquaglia90**
# This Pull request:
I did a local update of root/master with this, i was not able to compile locally on my mac for few other reasons. Suggestions on changes and checks to add welcome. 

## Changes or fixes:
Add the CDF calls from Math/ProbFunc  to speed up the integrals of RooLandau PDFs. 
Code has been used in on-going analysis at LHCb, with significant speed-up 

## Checklist:

- [ ] tested changes locally
- [ ] updated the docs (if necessary)

This PR fixes #

### CODE REVIEW

**Comment by hageboeck** [comment]
@rquaglia90 this is great!

If you have the time, could you add a test of analytic vs numeric integral? It should be sufficient to copy-paste the Johnson test code, and run it for a few parameter points (not many needed) from here:
https://github.com/root-project/root/blob/7b8424497d2c6e5494f8de225c1e2b315a01e6d3/roofit/roofit/test/testRooJohnson.cxx#L101-L182

**Comment by rquaglia90** [comment]
Sure, and actually I am interested in having such kind of unit tests also in some custom pdfs I coded up and plan to push to Root if I have some free time.
I first have to get the root master compiled on my MacBook to run the unit tests or are those tests something which runs in the CI? Or are those tests something which runs as macros?

**Comment by hageboeck** [comment]
> Sure, and actually I am interested in having such kind of unit tests also in some custom pdfs I coded up and plan to push to Root if I have some free time.
> I first have to get the root master compiled on my MacBook to run the unit tests or are those tests something which runs in the CI? Or are those tests something which runs as macros?

The first two, actually. If you register the unit test in `roofit/roofit/test/CMakeLists.txt`, it will run with the CI. You can get a lot of inspiration from the existing tests on how to do that!
I also recommend to run them on your macbook by compiling yourself, as the turnaround time is *much* faster.
Run them either by compiling and directly starting the exectuables under `<buildDirectory>/roofit/roofit/tests/<executable>` or `ctest -R testName -V`.

File: roofit/roofit/src/RooLandau.cxx [hunk 1]
```diff
@@ -72,7 +73,24 @@ void RooLandau::computeBatch(cudaStream_t* stream, double* output, size_t nEvent
...
+}
+
+Double_t RooLandau::analyticalIntegral(Int_t code, const char *rangeName) const
+{
+   assert(code == 1);
+
+   double max = x.max(rangeName);
+   double min = x.min(rangeName);
+
+   return ROOT::Math::landau_cdf(max, sigma, mean) - ROOT::Math::landau_cdf(min, sigma, mean);
```
Comment (tcuisset) [inline roofit/roofit/src/RooLandau.cxx:93]: You have to multiply by sigma the result of `RooLandau::analyticalIntegral` as `RooLandau::evaluate` uses `TMath::Landau()` with `norm=kFalse`

Comment (guitargeek) [reply roofit/roofit/src/RooLandau.cxx]: Thanks for spotting this! You are right of course

**Comment by guitargeek** [comment]
@phsft-bot build

**Review by guitargeek — APPROVED** [review]
LGTM, thank you very much for this effective improvement!

To get this into the 6.28 release, I addressed the remaining comments myself, so this can be merged quickly. I added the missing sigma factor as pointed out by @tcuisset, and implemented the unit test as suggested by @hageboeck.

I also had to make some changes to the tutorials about numeric integration, because these relied on the fact that `RooLandau` doesn't support analytic integration.

Then, as usual, I added some commits with general code improvements.

**Comment by guitargeek** [comment]
@phsft-bot build

**Pull request merged**